tests/phpunit/Configuration/Entry/BadgeTest.php

```diff
@@ -36,16 +36,54 @@
 namespace Infection\Tests\Configuration\Entry;
 
 use Infection\Configuration\Entry\Badge;
+use InvalidArgumentException;
 use PHPUnit\Framework\TestCase;
 
 final class BadgeTest extends TestCase
 {
-    use BadgeAssertions;
+    /** @dataProvider branch_names_to_be_matched */
+    public function test_branch_match(string $branchName, string $branchMatch, bool $willMatch): void
+    {
+        $this->assertSame(
+            $willMatch,
+            (new Badge($branchMatch))
+                ->applicableForBranch($branchName)
+        );
+    }
+
+    /** @return non-empty-list<array{string, non-empty-string, bool}> */
+    public function branch_names_to_be_matched(): array
+    {
+        return [
+            ['master', 'master', true],
+            ['main', 'main', true],
+            ['main', 'master', false],
+            ['mast', 'master', false],
+            ['master ', 'master', false],
+            [' master', 'master', false],
+            [' master ', 'master', false],
+            ['master1', 'master', false],
+            ['foo', '/^(foo|bar)$/', true],
+            ['bar', '/^(foo|bar)$/', true],
+            ['foobar', '/^(foo|bar)$/', false],
+            ['fo', '/^(foo|bar)$/', false],
+            ['ba', '/^(foo|bar)$/', false],
+            ['foo ', '/^(foo|bar)$/', false],
+            [' foo', '/^(foo|bar)$/', false],
+            ['foo1', '/^(foo|bar)$/', false],
```

Comment:
You can also do `yield 'test name' => ['master', 'master', true];` but I'm nitpicking here.

Reply:
I prefer to avoid `Generator` when a simpler type is sufficient :+1: